src/puppetlabs/puppetdb/queue.clj

```diff
@@ -144,5 +153,11 @@
   (count [this]
     (.size fifo-queue)))
 
-(defn sorted-command-buffer [^long n]
-  (SortedCommandBuffer. (TreeMap.) (HashMap.) n))
+(defn sorted-command-buffer
+  ([^long n]
+   (sorted-command-buffer n (constantly nil)))
+  ([^long n ^clojure.lang.IFn delete-update-fn]
+   ;; accepting a function here is a hack to get around a cyclic dependency
+   ;; between this ns, mq-listener.clj, and dlo.clj. My hope is we'll be able
+   ;; to get rid of it somehow when we refactor mq-listener and command.clj.
+   (SortedCommandBuffer. (TreeMap.) (HashMap.) n delete-update-fn)))
```

Comment:
Bummer. We'll fix this up soon. Another thing to think about here, is much of the information that we're looking for here is already known in the buffer. We don't currently have a threadsafe way to do this, but, if we did, you could just ask for the size of the buffer and you'd know the number of enqueue (as an example). Similarly, walking the datastructure could get the number of invalidated.